components/brave_new_tab_ui/containers/newTab/gridPageButtons.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react'
+import { useCallback, useRef, useEffect } from 'react'
+import styled from 'styled-components'
+
+const PageIndicator = <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
+    <circle cx="50" cy="50" r="50" />
+</svg>
+
+const ListPageButtonContainer = styled('div') <{}>`
+  display: flex;
+  flex-direction: row;
+
+  --list-page-button-size: 8px;
+  gap: var(--list-page-button-size);
+`
+
+const StyledButton = styled('button')`
+    all: unset;
+
+    cursor: pointer;
+
+    color: var(--brave-palette-white);
+    opacity: 0.5;
+
+    :hover {
+        opacity: 0.8
+    }
+
+    width: var(--list-page-button-size);
+    height: var(--list-page-button-size);
+`
+
+interface GridPageButtonProps {
+    page: number
+    pageContainerRef: React.MutableRefObject<HTMLDivElement | undefined>
+}
+
+function GridPageButton (props: GridPageButtonProps) {
+    const updatePage = useCallback(() => props
+        .pageContainerRef
+        .current
+        ?.children[props.page]
+        .scrollIntoView({ behavior: 'smooth' }),
+        [props.page, props.pageContainerRef])
```

Comment:
This is interesting that you're having the button component itself control the pagination directly, instead of e.g. set a "page number" state variable and then have the pages observe that and change themselves when that variable is modified.
I think that's fine, I don't see a reason to change it for this. It's nice that all pagination logic is kind of in one place. But if we needed to for example, have it remember which page was scrolled to across NTP loads, then maybe that logic should be in the pages container itself, although I'm sure it could be done here too as long as the timing is right.

Reply:
Yeah, I did consider doing something like that but I figured until we need that it's nice to have everything related to scrolling encapsulated in one place. Happy to change if you prefer (we could even pass in a `scrollToPage` function here instead).